fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -1232,6 +1249,18 @@ ACTOR Future<Void> dataDistributionRelocator(DDQueueData* self,
 				while (tciIndex < self->teamCollections.size()) {
 					if (CLIENT_KNOBS->SHARD_ENCODE_LOCATION_METADATA && rd.isRestore()) {
 						auto req = GetTeamRequest(tciIndex == 0 ? rd.dataMove->primaryDest : rd.dataMove->remoteDest);
+
+						if (CLIENT_KNOBS->SHARD_ENCODE_LOCATION_METADATA && tciIndex == 1 && rd.dataMoveIdSetDelayed) {
```

Comment:
Here we are restoring a data move, the teams are determined already, hence I feel this block may not be necessary.